Gems/ROS2/Code/Source/EffortSensor/ROS2EffortSensorComponent.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2EffortSensorComponent.h"
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+#include <AzFramework/Physics/RigidBodyBus.h>
+#include <ROS2/Frame/ROS2FrameComponent.h>
+#include <ROS2/ROS2Bus.h>
+#include <ROS2/Utilities/ROS2Conversions.h>
+#include <ROS2/Utilities/ROS2Names.h>
+#include <AzFramework/Physics/SimulatedBodies/RigidBody.h>
+#include <Source/RigidBodyComponent.h>
+#include <PhysX/Joint/PhysXJointRequestsBus.h>
+namespace ROS2
+{
+    namespace Internal
+    {
+        const char* kWrenchMsgType = "geometry_msgs::msg::WrenchStamped";
+    }
+
+    void ROS2EffortSensorComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<ROS2EffortSensorComponent, ROS2SensorComponent>()->Version(1);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<ROS2EffortSensorComponent>("ROS2 Effort Sensor", "Effort sensor component")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"));
+            }
+        }
+    }
+
+    AZ::Entity::ComponentArrayType ROS2EffortSensorComponent::findCompatibleJointComponents(){
+        AZ::Entity::ComponentArrayType components;
+        for (auto type : CompatibleJointComponents ){
+            auto foundComponents = m_entity->FindComponents(type);
+            AZStd::move(foundComponents.begin(), foundComponents.end(), AZStd::back_inserter(components));
+        }
+        return components;
+    }
+
+    void ROS2EffortSensorComponent::Activate()
+    {
+        AZ::Entity::ComponentArrayType componentPtrVec = findCompatibleJointComponents();
+        AZ_Warning("ROS2EffortSensorComponent", componentPtrVec.size()>0, "ROS2EffortSensorComponent can work with only one joint component.");
```

Comment:
nit: It's generally preferred to use `empty()` or `!empty()` as opposed to comparing the container size to 0 (as in some (weird) cases it can be a performance 'pesimization')